tests/src/python/test_qgsprofilesourceregistry.py

```diff
@@ -0,0 +1,395 @@
+        layers = exporter.toLayers()
+        self.assertEqual(len(layers), 1)
+        layer = layers[0]
+        self.assertEqual(layer.wkbType(), Qgis.WkbType.PointZ)
+        self.assertEqual(layer.crs(), QgsCoordinateReferenceSystem("EPSG:2056"))
+        self.assertEqual(layer.featureCount(), 5)
+
+        expected_values = {
+            0: 454.8,
+            2199.9: 429.3,
+            4399.9: 702.5,
+            6430.1: 857.9,
+            8460.4: 1282.7
+        }
+        expected_z = list(expected_values.values())
+        for i, feature in enumerate(layer.getFeatures()):
+            z = feature.geometry().constGet().z()
+            self.assertEqual(z, expected_z[i])
+            self.assertEqual(z, feature["elevation"])
+            self.assertIn(feature["distance"], expected_values)
+            self.assertEqual(expected_values[feature["distance"]], feature["elevation"])
+
+    def test_profile_canvas_custom_source(self):
+        canvas = QgsElevationProfileCanvas()
+        canvas.setProject(QgsProject.instance())
+        canvas.setCrs(QgsCoordinateReferenceSystem("EPSG:2056"))
+        curve = QgsLineString()
+        curve.fromWkt("LINESTRING (2584085.816 1216473.232, 2579969.567 1218027.326, 2576819.377 1220589.481)")
+        canvas.setProfileCurve(curve.clone())
+        spy = QSignalSpy(canvas.activeJobCountChanged)
+        self.assertTrue(spy.isValid())
+
+        source = MyProfileSource()
+        QgsApplication.profileSourceRegistry().registerProfileSource(source)
+        canvas.refresh()
+        spy.wait()
+
+        distance_range = canvas.visibleDistanceRange()
+        self.assertTrue(distance_range.contains(0), f"Distance 0 (min) not included in range ({distance_range})")
+        self.assertTrue(distance_range.contains(8460.4), f"Distance 8460.4 (max) not included in range ({distance_range})")
+
+        elevation_range = canvas.visibleElevationRange()
+        self.assertTrue(elevation_range.contains(429.3), f"Elevation 429.3 (min) not included in range ({elevation_range})")
+        self.assertTrue(elevation_range.contains(1282.7), f"Elevation 1282.7 (max) not included in range ({elevation_range})")
+        QgsApplication.profileSourceRegistry().unregisterProfileSource(source)
+
+    def test_layout_item_profile_custom_source(self):
+        """
+        Test getting a custom profile in a layout item
+        """
+        source = MyProfileSource()
+        QgsApplication.profileSourceRegistry().registerProfileSource(source)
+
+        layout = QgsLayout(QgsProject.instance())
+        layout.initializeDefaults()
+
+        profile_item = QgsLayoutItemElevationProfile(layout)
+        layout.addLayoutItem(profile_item)
+        profile_item.attemptSetSceneRect(QRectF(10, 10, 180, 180))
+
+        curve = QgsLineString()
+        curve.fromWkt(
+            "LINESTRING (2584085.816 1216473.232, 2579969.567 1218027.326, 2576819.377 1220589.481)")
+
+        profile_item.setProfileCurve(curve)
```

Comment:
You'll need to set the text format for the plot to use the standard QGIS test font, at at least 16 points and bold for this test to work nicely across different platforms

Reply:
Thanks for the tip, applied in https://github.com/qgis/QGIS/pull/57311/commits/625fdee576f43f06180504f1c2df16bb13acd7a1